lib/Ix/DBIC/ResultSet.pm

```diff
@@ -515,6 +516,15 @@ sub _ix_wash_rows ($self, $rows) {
     for my $key ($by_type{boolean}->@*) {
       $row->{$key} = $row->{$key} ? $true : $false if defined $row->{$key};
     }
+
+    for my $key ($by_type{datetime}->@*) {
+      if ($row->{$key}) {
+        # Doesn't already look like an RFC3339 Zulu date?
+        if ($row->{$key} !~ /Z/) {
+          $row->{$key} = parsepgdate($row->{$key});
+        }
+      }
+    }
```

Comment:
This method seems to get data from three places -
 * User input
 * Defaults
 * The database

User input is always turned into an Ix::DateTime object, so we can trust that to look right (I think).

Defaults, now that I look at them more, could present a problem. They aren't validated against _ix_check_user_properties, so something unexpected could be in there. What do we do about that? (For example, the string "now()" is perfectly valid as far as the database is concerned!)

Since the database is storing a timestamptz, we *should* always get something back that DateTime::Format::Pg should understand, (or null which we ignore), so that should be fine.